torch/_inductor/ir.py

```diff
@@ -2105,6 +2106,123 @@ def realize(self):
         pass
 
 
+class BufferList(IRNode):
+    def __init__(self, data):
+        super().__init__()
+        self.name = V.graph.register_buffer(self)
+        self.buffers = [
+            ListElemBuffer(layout, self, i) for i, layout in enumerate(data.layouts)
+        ]
+        names = [t.name for t in self.buffers]
+        self.list_name = V.graph.register_list(names)
+        self.data = data
+
+    def __getitem__(self, ind):
+        return TensorBox.create(self.buffers[ind])
+
+    def make_indexer(self, ind):
+        return self.data.layouts[ind].make_indexer()
+
+    def get_name(self):
+        return self.name
+
+    def make_loader(self):
+        def fn():
+            load = self.data.make_loader()
+            return load()
+
+        return fn
+
+    def get_store_function(self):
+        def fn(result):
+            ops.store(self.list_name, self.data.layouts, result)
+
+        return fn
+
+    def decide_layout(self):
+        pass
+
+    def get_device(self):
+        return self.data.layouts[0].device
+
+    def get_layouts(self):
+        return self.data.layouts
+
+    def is_no_op(self):
+        return False
+
+    def get_alias_names(self):
+        return ()
+
+    def get_mutation_names(self):
+        return ()
+
+    def list_arg_count(self):
+        _, (list_reads, list_writes) = self._list_read_writes()
+        return len(list_reads) + len(list_writes)
+
+    @cache_on_self
+    def _list_read_writes(self):
+        def fn():
+            self.get_store_function()(self.make_loader()())
+
+        return dependencies.extract_list_read_writes(fn)
+
+    def get_read_writes(self):
+        read_writes, _ = self._list_read_writes()
+        return read_writes
+
+
+class ListElemBuffer(Buffer):
```

Comment:
This seems like a duplicate of MultiOutput?

Reply:
I can possibly subclass that I think. I'll try this tm

Reply:
Removed